### PR TITLE
Fix build with recursion-schemes >= 5.2

### PR DESCRIPTION
**Pull request by hasufell**
Additionally we *could* bump the index state once https://github.com/input-output-hk/cardano-prelude/pull/122 is merged

### CODE REVIEW

File: binary/src/Cardano/Binary/ToCBOR.hs [hunk 1]
```diff
@@ -166,7 +172,11 @@ instance B.Buildable t => B.Buildable (SizeF t) where
           TodoF _ x -> bprint ("(_ :: " . shown . ")") (typeRep x)
 
 instance B.Buildable (Fix SizeF) where
+#if MIN_VERSION_recursion_schemes(5,2,0)
+  build x = bprint build (unFix x)
+#else
   build x = bprint build (unfix x)
```
Comment (phadej) [inline binary/src/Cardano/Binary/ToCBOR.hs:178]: fwiw, `project = unfix` for `Fix`.

**Comment by hasufell** [comment]
ping

**Comment by nc6** [comment]
Looks generally good. I guess I tend to prefer no CPP, so what would happen if we just require the newer version? Would this cause problems?

**Comment by hasufell** [comment]
> Would this cause problems?

I tend to be defensive and since I cannot say "no", I assume it will.

**Comment by mark-stopka** [comment]
This olddated dependency is actually causing problems for us at https://github.com/input-output-hk/cardano-node/issues/1997, I have already prepared a DRAFT PR for version downgrade https://github.com/opensuse-haskell/configuration/pull/31...

I am going to try your patch now...

**Comment by mark-stopka** [comment]
I see it's against outdated branch that does not contain the constraints, would someone please be so kind and rebase on current master?
 
CC: @mimi1vx

**Comment by mark-stopka** [comment]
I merged master into this feature branch on my own fork and all test pass, can we please get upstream up-to-date, this is really blocking us in https://github.com/input-output-hk/cardano-node/issues/1997 and https://github.com/opensuse-haskell/configuration/pull/31, @dcoutts?

**Comment by hasufell** [comment]
@mark-stopka I updated https://github.com/hasufell/cardano-base/tree/jospald/PR/fix-build-redcursion-schemes-5.2

**Comment by mark-stopka** [comment]
Thank you @hasufell!

**Comment by mark-stopka** [comment]
@hasufell, I can still see your PR does not address wollowing LOC...

https://github.com/input-output-hk/cardano-base/blob/6012e2fcbddc516490fcae07aad8d3e96fd729e3/binary/cardano-binary.cabal#L48

**Comment by mark-stopka** [comment]
Is this going to be merged into upstream? We have a draft work-around we would like to dispose of...

https://github.com/opensuse-haskell/configuration/pull/31

**Comment by mark-stopka** [comment]
@dcoutts is there anything blocking this from being merged into upstream? We have out of three patch, but we have "upstream first" policy... :)

**Comment by mark-stopka** [comment]
@dcoutts, @erikd, any idea what is blocking this and if @mimi1vx can help? I myself am busy with Rust implementation of Cardano networking protocols... and few other projects...

- [2nd-Layer/rust-cardano-ouroboros-network](https://github.com/2nd-Layer/rust-cardano-ouroboros-network)

**Comment by newhoggy** [comment]
What's the story with this PR?  Can it be merged?

**Pull request closed without merging**